### PR TITLE
java.lang.IndexOutOfBoundsException: Source emitted more than one item

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`Caused by: java.lang.IllegalStateException: no known node for /home/jenkins/build/workspace/934`

test this please

**Comment by chibenwa** [comment]
(BTW does it solves https://github.com/linagora/james-project/pull/3242 ?)

**Comment by rouazana** [comment]
> (BTW does it solves #3242 ?)

no it does not! This stack trace is visible in #3242, but fixing it does not fix #3242.

**Comment by chibenwa** [comment]
```
[98f3e9d906d49cec40c18d3e892b2ddba9203e4f] [ERROR] src/main/java/org/apache/james/mailbox/store/mail/MessageIdMapper.java:[23,8] (imports) UnusedImports: Unused import - java.util.Map.
```

**Comment by Arsnael** [comment]
```
 [a9475d47fb1ae97215b34943b6f803fd73c27f9d] [INFO] There are 2 errors reported by Checkstyle 8.29 with checkstyle.xml ruleset.
[a9475d47fb1ae97215b34943b6f803fd73c27f9d] [ERROR] src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java:[24,8] (imports) UnusedImports: Unused import - java.util.Map.
[a9475d47fb1ae97215b34943b6f803fd73c27f9d] [ERROR] src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java:[26,8] (imports) UnusedImports: Unused import - java.util.function.BiFunction.
[...]
[a9475d47fb1ae97215b34943b6f803fd73c27f9d] [ERROR] src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMessageIdMapper.java:[26,8] (imports) UnusedImports: Unused import - java.util.Map.
[a9475d47fb1ae97215b34943b6f803fd73c27f9d] [ERROR] src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMessageIdMapper.java:[27,8] (imports) UnusedImports: Unused import - java.util.function.BinaryOperator.
[a9475d47fb1ae97215b34943b6f803fd73c27f9d] [ERROR] src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMessageIdMapper.java:[29,8] (imports) UnusedImports: Unused import - java.util.stream.Collectors.
[a9475d47fb1ae97215b34943b6f803fd73c27f9d] [ERROR] src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMessageIdMapper.java:[50,8] (imports) UnusedImports: Unused import - com.google.common.collect.ImmutableMap.
```

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java [hunk 1]
```diff
@@ -205,14 +204,14 @@ public void delete(MessageId messageId) {
     }
 
     @Override
-    public Map<MailboxId, UpdatedFlags> setFlags(MessageId messageId, List<MailboxId> mailboxIds, Flags newState, MessageManager.FlagsUpdateMode updateMode) throws MailboxException {
+    public Multimap<MailboxId, UpdatedFlags> setFlags(MessageId messageId, List<MailboxId> mailboxIds, Flags newState, MessageManager.FlagsUpdateMode updateMode) throws MailboxException {
```
Comment (mbaechler) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java:207]: I may deserve a comment to explain in which case you have several UpdatedFlags per MailboxId

**Comment by chibenwa** [comment]
```
[cd6eedfc4a8280392285ff8cecee9b52fe0adac3] [INFO] Compiling 3 source files to /james-project/event-sourcing/event-store-api/target/classes at 1585736186787
[cd6eedfc4a8280392285ff8cecee9b52fe0adac3] [ERROR] #
[cd6eedfc4a8280392285ff8cecee9b52fe0adac3] [ERROR] #OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x0000000633c00000, 484442112, 0) failed; error='Not enough space' (errno=12)
[cd6eedfc4a8280392285ff8cecee9b52fe0adac3] [INFO] # There is insufficient memory for the Java Runtime Environment to continue.
[cd6eedfc4a8280392285ff8cecee9b52fe0adac3] [INFO] # Native memory allocation (mmap) failed to map 484442112 bytes for committing reserved memory.
[cd6eedfc4a8280392285ff8cecee9b52fe0adac3] [ERROR] # An error report file with more information is saved as:
[cd6eedfc4a8280392285ff8cecee9b52fe0adac3] [INFO] # /james-project/hs_err_pid1462.log
```

Can be considered :green_apple: 

Running on ci-james-04 in /home/jenkins/build/workspace/workflow

**Comment by chibenwa** [comment]
```
[f99b61f289ca65336df9d7bd5059022a17136dde] Exception in thread "Thread-1642" java.lang.RuntimeException: java.io.IOException: Failed to bind to /0.0.0.0:43321
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.apache.james.webadmin.WebAdminServer.lambda$start$1(WebAdminServer.java:95)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at spark.Service.lambda$init$2(Service.java:582)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at java.base/java.lang.Thread.run(Thread.java:834)
[f99b61f289ca65336df9d7bd5059022a17136dde] Caused by: java.io.IOException: Failed to bind to /0.0.0.0:43321
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:346)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:308)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:236)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.eclipse.jetty.server.Server.doStart(Server.java:396)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at spark.embeddedserver.jetty.EmbeddedJettyServer.ignite(EmbeddedJettyServer.java:149)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at spark.Service.lambda$init$2(Service.java:574)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	... 1 more
[f99b61f289ca65336df9d7bd5059022a17136dde] Caused by: java.net.BindException: Address already in use
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at java.base/sun.nio.ch.Net.bind0(Native Method)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at java.base/sun.nio.ch.Net.bind(Net.java:461)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at java.base/sun.nio.ch.Net.bind(Net.java:453)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:342)
[f99b61f289ca65336df9d7bd5059022a17136dde] 	... 9 more
```

test this please

**Comment by chibenwa** [comment]
```
[f99b61f289ca65336df9d7bd5059022a17136dde] docker: Error response from daemon: No such container: git-publish-mailboxdeleteduringupdateexception-c01232e7-88b9-4a57-b668-f79cbec834cf.
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**